src/linear_algebra/projective_space/subspace.lean

```diff
@@ -185,6 +185,160 @@ lemma span_eq_span_iff {S T : set (ℙ K V)} : span S = span T ↔ S ⊆ span T
+      { rw [ha, hb, zero_add], simp },
+      { simp_rw [ha, zero_add], right, exact hb },
+      { simp_rw [hb, add_zero], right, exact ha },
+      { right, cases ha, cases hb, exact ⟨hab, mem_add W a b ha_w hb_w hab ha_h hb_h⟩ } } },
+  { simp },
+  { intros i hi, exact hf i hi }
+end
+
+/-- If a family of vectors is such that every nonzero vector in the family determines a point in the
+corresponding projective space which is contained in a subspace, then every nonzero linear
+combination of vectors from the family also determines a point contained in that subspace. -/
+lemma mk_sum_smul_mem {ι : Type*} (s : finset ι) (W : subspace K V) (f : ι → V) (g : ι → K)
+  (hf : ∀ i, i ∈ s → f i = 0 ∨ ∃ (hi : f i ≠ 0), projectivization.mk K (f i) (hi) ∈ W)
+  (hs : ∑ i in s, (g i) • f i ≠ 0) : projectivization.mk K (∑ i in s, (g i) • f i) hs ∈ W :=
+begin
+  refine mk_sum_mem s W (g • f) _ hs,
+  intro i,
+  by_cases hgz : g i = 0,
+  { rw [hgz, zero_smul], simp },
+  { by_cases hfz : f i = 0,
+    { rw [hfz, smul_zero], simp },
+    { intro hi, right,
+      refine ⟨by { rw [ne.def, smul_eq_zero, not_or_distrib], exact ⟨hgz, hfz⟩ }, _⟩,
+      cases (or.resolve_left (hf i hi) hfz), convert h using 1, rw mk_eq_mk_iff', use g i } }
+end
+
+/-- If a set of points in a projective space has the property that for any two unique points
+contained in the set, these points being dependent with a third point in the projective space
+implies that this third point is contained in the set, then the set is a subspace. -/
+def mk_of_dependent (S : set (ℙ K V))
+  (h : ∀ u v w, u ≠ v → u ∈ S → v ∈ S → dependent ![u, v, w] → w ∈ S) : subspace K V :=
+{ carrier := S,
+  mem_add' := λ u v hu hv huv huS hvS,
+  begin
+    by_cases heq : projectivization.mk K u hu = projectivization.mk K v hv,
+    { convert hvS using 1,
+      rw mk_eq_mk_iff' at heq ⊢,
+      cases heq with a ha,
+      exact ⟨a + 1, by { rw [add_smul, ha, one_smul] }⟩ },
+    { refine h _ _ (projectivization.mk K (u + v) huv) heq huS hvS _,
+      convert dependent.mk ![u, v, u + v] _ _,
+      { ext i, fin_cases i; simp },
+      { intro i, fin_cases i; assumption },
+      { rw fintype.not_linear_independent_iff,
+        refine ⟨![-1, -1, 1], _, ⟨2, by { simp }⟩⟩,
+        simp only [fin.sum_univ_three, matrix.cons_val_zero, neg_smul, one_smul,
+          matrix.cons_val_one, matrix.head_cons, matrix.cons_vec_bit0_eq_alt0, matrix.cons_append,
+          matrix.empty_append, matrix.cons_vec_alt0],
+        abel } },
+  end }
+
+/-- If a set of points in a projective space has the property that for any independent family of
+points contained in the set, this family being dependent with another point in the projective space
+implies that this point is contained in the set, then the set is a subspace. -/
+def mk_of_dependent' (S : set (ℙ K V)) (h : ∀ (ι : Type*) (f : ι → ℙ K V) (hf: independent f)
+  (u : ℙ K V) (hf' : dependent (λ t, sum.rec_on t f (λ _, u) : ι ⊕ punit → ℙ K V))
```

Comment:
Any reason you used `ι ⊕ punit` instead of `option ι` here? The latter feels more canonical.